### PR TITLE
[net-diag] add TLV for MLE counters

### PR DESCRIPTION
**Pull request by abtink**
This commit introduces Network Diagnostic TLV for MLE counters.
It updates `NetworkDiagnostic::Server` to provide this TLV when
queried and provide support for retrieving on `Client`.

----

This PR contains the commit from https://github.com/openthread/openthread/pull/8866

Related to [SPEC-1186](https://threadgroup.atlassian.net/browse/SPEC-1186).

### CODE REVIEW

File: src/cli/cli.cpp [hunk 1]
```diff
@@ -5798,6 +5798,122 @@ template <> otError Interpreter::Process<Cmd("meshdiag")>(Arg aArgs[])
...
+
+        SuccessOrExit(error = otMeshDiagQueryChildTable(GetInstancePtr(), routerRloc16,
+                                                        HandleMeshDiagQueryChildTableResult, this));
+
+        error = OT_ERROR_PENDING;
+    }
+    /**
+     * @cli meshdiag childip6
+     * @code
+     * meshdiag childrenip6addr 0xdc00
```
Comment (Vyrastas) [inline src/cli/cli.cpp:5853]: childrenip6addr --> childip6

File: src/cli/cli.cpp [hunk 2]
```diff
@@ -5798,6 +5798,122 @@ template <> otError Interpreter::Process<Cmd("meshdiag")>(Arg aArgs[])
...
+     *     rx-on:yes type:ftd full-net:yes
+     *     rss - ave:-20 last:-20 margin:80
+     *     err-rate - frame:11.51% msg:0.76%
+     *     csl - sync:no period:0 timeout:0 channel:0
+     * rloc16:0x6403 ext-addr:ee24e64ecf8c079a ver:4
+     *     timeout:120 age:19 supvn:129 q-msg:0
+     *     rx-on:no type:mtd full-net:no
+     *     rss - ave:-20 last:-20  margin:80
+     *     err-rate - frame:0.73% msg:0.00%
+     *     csl - sync:no period:0 timeout:0 channel:0
```
Comment (Vyrastas) [inline src/cli/cli.cpp:5816]: Need `Done` at the end?

File: src/cli/cli.cpp [hunk 3]
```diff
@@ -5798,6 +5798,122 @@ template <> otError Interpreter::Process<Cmd("meshdiag")>(Arg aArgs[])
...
+     * @code
+     * meshdiag childrenip6addr 0xdc00
+     * child-rloc16: 0xdc02
+     *     fdde:ad00:beef:0:ded8:cd58:b73:2c21
+     *     fd00:2:0:0:c24a:456:3b6b:c597
+     *     fd00:1:0:0:120b:95fe:3ecc:d238
+     * child-rloc16: 0xdc03
+     *     fdde:ad00:beef:0:3aa6:b8bf:e7d6:eefe
+     *     fd00:2:0:0:8ff8:a188:7436:6720
+     *     fd00:1:0:0:1fcf:5495:790a:370f
```
Comment (Vyrastas) [inline src/cli/cli.cpp:5861]: Need `Done` at the end?

File: src/cli/cli.cpp [hunk 4]
```diff
@@ -5798,6 +5798,122 @@ template <> otError Interpreter::Process<Cmd("meshdiag")>(Arg aArgs[])
...
+     * @code
+     * meshdiag neighbortable 0x7400
+     * rloc16:0x9c00 ext-addr:764788cf6e57a4d2 ver:4
+     *    rx-on:yes type:ftd full-net:yes
+     *    rss - ave:-20 last:-20 margin:80
+     *    err-rate - frame:1.38% msg:0.00%
+     * rloc16:0x7c00 ext-addr:4ed24fceec9bf6d3 ver:4
+     *    rx-on:yes type:ftd full-net:yes
+     *    rss - ave:-20 last:-20 margin:80
+     *    err-rate - frame:0.72% msg:0.00%
```
Comment (Vyrastas) [inline src/cli/cli.cpp:5891]: Need `Done` at the end?

File: src/cli/cli.cpp [hunk 5]
```diff
@@ -5798,6 +5798,122 @@ template <> otError Interpreter::Process<Cmd("meshdiag")>(Arg aArgs[])
...
+     * * Extended MAC address
+     * * Thread Version
+     * * Timeout (in seconds)
+     * * Age (seconds since last heard)
+     * * Supervision interval (in seconds)
+     * * Number of queued messages (in case child is sleepy)
+     * * Device Mode
+     * * RSS (average and last)
+     * * Error rates: frame tx (at MAC layer), IPv6 message tx (above MAC)
+     * * CSL info
```
Comment (Vyrastas) [inline src/cli/cli.cpp:5831]: colon at end

CSL info:

**Review by Vyrastas — CHANGES_REQUESTED** [review]

**Comment by abtink** [comment]
Thanks @Vyrastas. I added the suggestions above in https://github.com/openthread/openthread/pull/8866#issuecomment-1540973549 (which is the main PR adding "Child TLV and Neighbor TLV" for Network Diagnostic, etc).

This PR is for MLE Counters TLV (but I added it on top of the commit from the other PR to avoid conflict when we merge them). I may later include this in #8866 and close this PR.

**Comment by abtink** [comment]
I include the commit in this PR into https://github.com/openthread/openthread/pull/8866. So we can close this.

**Pull request closed without merging**